src/app/portlets/content-types/fields/content-type-fields-add-row/content-type-fields-add-row.component.spec.ts

```diff
@@ -0,0 +1,269 @@
+import { ContentTypeFieldsAddRowComponent } from './content-type-fields-add-row.component';
+import { MessageService } from './../../../../api/services/messages-service';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { TooltipModule } from 'primeng/primeng';
+import { async, ComponentFixture, fakeAsync, tick } from '@angular/core/testing';
+import { DOTTestBed } from '../../../../test/dot-test-bed';
+import { DebugElement, Component, Input, Output, EventEmitter, ViewChild, ElementRef } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { Field } from '../';
+import { Observable } from 'rxjs/Observable';
+import { HotkeysService, Hotkey } from 'angular2-hotkeys';
+import { TestHotkeysMock } from './../../../../test/hotkeys-service.mock';
+import { MockMessageService } from './../../../../test/message-service.mock';
+
+describe('ContentTypeFieldsAddRowComponent', () => {
+    let comp: ContentTypeFieldsAddRowComponent;
+    let fixture: ComponentFixture<ContentTypeFieldsAddRowComponent>;
+    let de: DebugElement;
+    let el: HTMLElement;
+    let testHotKeysMock: TestHotkeysMock;
+    let columns: number[];
+    let disabled: boolean;
+    let toolTips: string[];
+
+    const messageServiceMock = new MockMessageService({
+        'contenttypes.content.add_rows': 'Add Rows',
+        'contenttypes.content.one_column': 'One column',
+        'contenttypes.content.two_columns': 'Second column',
+        'contenttypes.content.three_columns': 'Three columns',
+        'contenttypes.content.four_columns': 'Four columns'
+    });
+
+    beforeEach(() => {
+        testHotKeysMock = new TestHotkeysMock();
+
+        DOTTestBed.configureTestingModule({
+            declarations: [
+                ContentTypeFieldsAddRowComponent
+            ],
+            imports: [TooltipModule, BrowserAnimationsModule],
+            providers: [
+                { provide: HotkeysService, useValue: testHotKeysMock },
+                { provide: MessageService, useValue: messageServiceMock}
+            ]
+        });
+
+        fixture = DOTTestBed.createComponent(ContentTypeFieldsAddRowComponent);
+        de = fixture.debugElement;
+        comp = fixture.componentInstance;
+        el = de.nativeElement;
+    });
+
+    it('should render disabled input', () => {
+        disabled = true;
+        comp.disabled = disabled;
+
+        fixture.detectChanges();
+
+        const buttonElement = de.query(By.css('button'));
+
+        expect(comp.disabled).toEqual(buttonElement.nativeElement.disabled);
```

Comment:
So in the expects we should always do:

```expect(THING_WE_ARE_TESTING).to(OUR_TEST_VALUE)```

So in this case:

```expect(buttonElement.nativeElement.disabled).toEqual(true);``` <-- because you know you are setting the ```true```.